Removed mention of Mercurial and tried to clarify getting and buildin…

jseldess opened the pull request:
…g steps

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/3486)

<!-- Reviewable:end -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cockroachdb/cockroach?pullRequest=3486) <br/>You should sign our Contributor License Agreement in order to get your pull request merged.<br/><hr/>**JESSE SELDESS** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please add the email address used for this commit to your account ([for further information, click here] (https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user)).

tamird commented:
@jseldess your git config seems to be a bit off. Can you follow https://help.github.com/articles/set-up-git/ and then `git commit --amend --reset-author && git push -f`?

tamird commented on CONTRIBUTING.md:
````diff
@@ -2,28 +2,26 @@
-You can `go get -d github.com/cockroachdb/cockroach` or, alternatively,
+    ```bash
+    go get -d github.com/cockroachdb/cockroach
+    cd cockroach
````
this should be `cd $GOPATH/src/github.com/cockroachdb/cockroach`

tamird commented on CONTRIBUTING.md:
````diff
@@ -2,28 +2,26 @@
-git clone git@github.com:cockroachdb/cockroach.git
-cd cockroach
-```
+    Alternately:
````
I would remove these instructions. Anyone that is hand-cloning the repo knows how to map the above `go get`-based instructions to this.

tamird commented on CONTRIBUTING.md:
```diff
@@ -2,28 +2,26 @@
-Now you should be all set for `make build`, `make test` and everything else our Makefile has to
-offer. Note that the first time you run `make` various dependent libraries and tools will be
-downloaded and installed which can be somewhat time consuming. Be patient.
+3.  Run `make build`, `make test`, or anything else our Makefile offers. Note that the first time you run `make`, it can take some time to download and install various dependent libraries and tools. 
```
replace "dependent libraries and tools" with "dependencies" - the current phrasing implies that they depend on us.

Commit:
Removed mention of Mercurial and tried to clarify getting and building steps

updating my git config

Updating my git config

Added a bit more guidance to step 1, added $GOPATH/src/github.com/cockroachdb/cockroach and removed the git clone option from step 2, and changed dependent library and tools to dependencies in step 3.

bdarnell commented:
LGTM, but you need to click the "CLA not signed yet" button to agree to our license terms.
